src/frontend/src/lib/components/convert/ConvertInputAmount.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { IconClose } from '@dfinity/gix-components';
+	import { debounce, isNullish, nonNullish } from '@dfinity/utils';
+	import { BigNumber } from '@ethersproject/bignumber';
+	import InputCurrency from '$lib/components/ui/InputCurrency.svelte';
+	import type { Token } from '$lib/types/token';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+
+	export let token: Token;
+	export let amount: number | undefined = undefined;
+	export let disabled: boolean | undefined = undefined;
+	export let customValidate: (userAmount: BigNumber | undefined) => void = () => {};
+	export let error: Error | undefined = undefined;
+	export let placeholder = '0.00';
+
+	const onReset = () => {
+		amount = undefined;
+	};
+
+	const validate = () => {
+		const parsedValue = !invalidAmount(amount)
+			? parseToken({
+					value: `${amount}`,
+					unitName: token.decimals
+				})
+			: undefined;
+
+		customValidate(parsedValue);
+	};
+
+	const debounceValidate = debounce(validate, 300);
+
+	$: amount, token.decimals, debounceValidate();
+</script>
+
+<div class="convert-input-amount text-3xl font-bold" class:error={nonNullish(error)}>
+	<InputCurrency
+		name="convert-amount"
+		bind:value={amount}
+		decimals={token.decimals}
+		{placeholder}
+		{disabled}
+	>
+		<svelte:fragment slot="inner-end">
+			{#if nonNullish(amount) && !disabled}
+				<button
+					on:click|preventDefault={onReset}
+					class={nonNullish(error) ? 'text-error' : 'text-aurometalsaurus'}
+					class:hidden={isNullish(amount) || disabled}
+				>
+					<IconClose />
+				</button>
+			{/if}
+		</svelte:fragment>
+	</InputCurrency>
+</div>
+
+<style lang="postcss">
```

Comment:
`postcss`?

Reply:
On the line 80, I'm applying a tailwind class (`@apply h-14`) to the input. Why tailwind and not CSS - just wanted to use the exact same height value as the "token-info" container. I think I saw another place in the code where "postcss" is used, but please let me know if you think there are some constraints here I'm not aware of.

<img width="492" alt="Screenshot 2024-11-12 at 10 18 18" src="https://github.com/user-attachments/assets/6b29803a-6c14-474f-bd11-9683f82029dd">

Reply:
It has to be `postcss`? It does not work as well with our standard `scss` annotation?

Reply:
You're right, it can be done with `scss`, I was confused by WebStorm - it doesn't recognise `apply` mixin when set to `lang="scss"`. Updated.